### ISSUE TITLE
[feature] new generator: "application generator"

### ISSUE THREAD

**Issue by fulara** [conan-io/conan#6382]
There is one huge feature that I miss in conan, or maybe I just did not understand how to achieve this sort of usage.

In our artifactory we have various of binaries packages, however they can only ever be used one at a time because of the need to load the environments.  

in cargo for example you have an ability to do cargo install 'package' and it will jump into ~/.cargo/bin and you can use the binaries without any additional steps. 
I do understand that there is a need for environments to run packages apps but I think it still could be possible to provide a generator that would mimic this simplicity my proposal is as follows:

Given application named 'potato-producer' a new generator would be available called lets say 'app_gen' 

after doing conan install -g app_gen ~/bin  potato-producer/1.0.0
It would create a:
1) directory ~/bin/potato-producer.dir
2) a script ~/bin/potato-producer 
Script would present itself as the target application, in the simplest aproach it could be something as simple as:
```
#!/bin/bash
source ~/bin/potato-producer.dir/activate.sh
source ~/bin/potato-producer.dir/activate_run.sh
~/bin/potato-producer "${@}"
```

This would allow to install various packages from conan and have them as usable binaries without any hussle of dealing with environments.

- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by czoido** [comment]
Hi @fulara,

Have you tried to use the `imports()` method in the conanfile.py?
Maybe you could define a imports like this:

```
def imports(self):
    dest = os.getenv("CONAN_IMPORT_PATH", "bin")
    self.copy("*", dst=dest, src="bin")
    self.copy("*.dylib*", dst=dest, src="lib")
```
and then:

`conan install . -e CONAN_IMPORT_PATH=the_path_of_your_choice`

Another option is to create a custom generator: https://docs.conan.io/en/latest/howtos/custom_generators.html

Could that work for you?

**Comment by fulara** [comment]
Hello @czoido 
with regards to `imports` this has a problem that it does not load the env variables, so you still need to load envs.

Yea I thought of writing custom_generator for that, but I was wondering whats the conan view on that, as I think its sort of feature that package managers do have.  
I mentioned that other package managers support that, installing packages and just have them, `cargo` has it, `npm` I think has it.  They are capable of installing binaries without a need to source environment.

Option would be to build everything statically, that would work with `imports` but its not something we are doing right now. maybe we can consider that.

**Comment by czoido** [comment]
Hi @fulara ,

Now I understand your problem more clearly. I think that would be a valid feature for Conan and that it could help more users, so we'll give it a thought.

Thanks a lot for the feedback!

**Comment by memsharded** [comment]
This would be solved now with 2.0 ``deployers`` and ``VirtualRunEnv`` generators in case env-vars are necessary. Closing as solved.
